pkg/cmd/roachtest/tests/restore.go

```diff
@@ -135,8 +135,6 @@ func registerRestore(r registry.Registry) {
 			m := c.NewMonitor(ctx)
 			dul := roachtestutil.NewDiskUsageLogger(t, c)
 			m.Go(dul.Runner)
-			hc := roachtestutil.NewHealthChecker(t, c, c.All())
-			m.Go(hc.Runner)
```

Comment:
If we are going to remove this, perhaps we could get a quick sign off from SQL and remove all uses of it? 

```
rg NewHealthChecker pkg/cmd/roachtest/
pkg/cmd/roachtest/tests/restore.go
138:			hc := roachtestutil.NewHealthChecker(t, c, c.All())
417:				hc := roachtestutil.NewHealthChecker(t, c, c.All())

pkg/cmd/roachtest/tests/import.go
138:		hc := roachtestutil.NewHealthChecker(t, c, c.All())
264:				hc := roachtestutil.NewHealthChecker(t, c, c.All())

pkg/cmd/roachtest/tests/cluster_to_cluster.go
934:				hc := roachtestutil.NewHealthChecker(t, c, rd.crdbNodes())

pkg/cmd/roachtest/roachtestutil/health_checker.go
40:// NewHealthChecker returns a populated HealthChecker.
41:func NewHealthChecker(t test.Test, c cluster.Cluster, nodes option.NodeListOption) *HealthChecker {
```

I'm a little worried that these recent failures are telling us _something_. But, it isn't clear this implementation gives us enough actionable info to be useful.

Reply:
Test eng validated that these errors come from gcp networking infra flakes [here](https://cockroachlabs.slack.com/archives/C023S0V4YEB/p1692394051143259). 

I didn't remove the whole package and the calls to import because:
- I wanted to merge this asap without waiting on other teams. I will ping sql-queries about the PR though.
- as Renato mentioned, there is potential to improve the health checker in a way that reduces flakiness and tests new things, though that seems like a larger investment that I don't think we need to conduct right now.

Reply:
👍 Fine with me.  One option would be to build a "healthcheck" workload that runs from the workload binary on the host machine so that it isn't subject to testrunner->gce network hiccups and is exposed to all the same issues the existing workloads are.